app/modules/integrations/integrations_service.py

```
@@ -1636,3 +1797,513 @@ async def save_integration(
+            tokens = await self.jira_oauth.exchange_code_for_tokens(
+                request.code, request.redirect_uri
+            )
+
+            access_token = tokens.get("access_token")
+
+            if not access_token:
+                raise Exception("Failed to obtain access token from OAuth exchange")
+
+            resources = await self.jira_oauth.get_accessible_resources(access_token)
+            if not resources:
+                raise Exception("No accessible Jira resources returned for this user")
```

Ruff (0.14.3)

Abstract `raise` to an inner function

(TRY301)